yjit/src/codegen.rs

```diff
@@ -5004,6 +5027,10 @@ fn gen_struct_aref(
         }
     }
 
+    if flags & VM_CALL_OPT_SEND != 0 {
+        handle_opt_send_shift_stack(asm, argc as i32, ctx);
+    }
```

Comment:
Why check this in multiple different places?

Reply:
Because you can't do the stack manipulation before side_exits. So I have to do this in every kind of send call.

Reply:
It's ok to do stack_push and stack_pop and read values from the stack before exiting, but it's not ok to write to the stack before doing so.

> ```
> railsbench  % in yjit from 83.4% to 83.6%          +0.2
> erubi_rails % in yjit from 85.1% to 85.3%.         +0.2
> eruby       % in yjit from 100%  to 100%.          +0.0
> ```
> 
> We do have an increase in exits for erubi, but we also have more compiled blocks. We used to compile 7787 now we compile 7819. Don't personally think it seems like a regression. But happy for anyone else to look at it closer.

If the percentage in YJIT increases then I think we're good 👍